src/Controls/src/Core/Platform/Android/Extensions/ButtonExtensions.cs

```diff
@@ -36,35 +36,24 @@ public static void UpdateContentLayout(this MaterialButton materialButton, Butto
 			{
 				var contentLayout = button.ContentLayout;
 
-				// IconPadding calls materialButton.CompoundDrawablePadding				
+				// IconPadding calls materialButton.CompoundDrawablePadding
 				// Which is why we don't have to worry about calling setCompoundDrawablePadding
 				// ourselves for our custom implemented IconGravityBottom
 				materialButton.IconPadding = (int)context.ToPixels(contentLayout.Spacing);
 
-				// For IconGravityTextEnd and IconGravityTextStart, setting the Icon twice
-				// is needed to work around the Android behavior that caused
-				// https://github.com/dotnet/maui/issues/11755
 				switch (contentLayout.Position)
```

Comment:
Instead of toggling the icon here, we are doing that as part of the measure. This makes the updates happen all together instead of here, and then it is better for maintenance.